src/main/java/io/logz/jmx2logzio/Jmx2Logzio.java

```diff
@@ -0,0 +1,80 @@
+package io.logz.jmx2logzio;
+
+import io.logz.jmx2logzio.Utils.HangupInterceptor;
+import io.logz.jmx2logzio.Utils.MetricsPipeline;
+import io.logz.jmx2logzio.Utils.Shutdownable;
+import io.logz.jmx2logzio.clients.JavaAgentClient;
+import io.logz.jmx2logzio.clients.JolokiaClient;
+import io.logz.jmx2logzio.configuration.Jmx2LogzioConfiguration;
+import io.logz.jmx2logzio.exceptions.IllegalConfiguration;
+import io.logz.jmx2logzio.objects.MBeanClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import static io.logz.jmx2logzio.configuration.Jmx2LogzioConfiguration.MetricClientType.JOLOKIA;
+import static io.logz.jmx2logzio.configuration.Jmx2LogzioConfiguration.MetricClientType.MBEAN_PLATFORM;
+
+public class Jmx2Logzio implements Shutdownable {
+
+    private static final Logger logger = LoggerFactory.getLogger(Jmx2Logzio.class);
+
+    private final Jmx2LogzioConfiguration conf;
+    private final ScheduledThreadPoolExecutor taskScheduler;
+    private final MBeanClient client;
+
+    public Jmx2Logzio(Jmx2LogzioConfiguration conf) {
+        this.conf = conf;
+
+        this.taskScheduler = new ScheduledThreadPoolExecutor(1);
+
+        if (conf.getMetricClientType() == JOLOKIA) {
+            this.client = new JolokiaClient(conf.getJolokiaFullUrl());
+            logger.info("Running with Jolokia URL: {}", conf.getJolokiaFullUrl());
+
+        } else if (conf.getMetricClientType() == MBEAN_PLATFORM) {
+            this.client = new JavaAgentClient();
+            logger.info("Running with Mbean client");
+        } else {
+            throw new IllegalConfiguration("Unsupported client type: " + conf.getMetricClientType());
```

Comment:
Originally, if the user used Jolokia agent, then he should have set the poller property in the configuration file. Since I changed that and wrote all the properties in the config file, the only case which it can happen is if the user will comment out the poller property in the conf file (and then it will be null). 
Is that too dummy proof? should I use ternary instead?

Reply:
@yyyogev Can you verify it's not null when you validate all the parameters? If so use trinary

Reply:
Will do